---
 mkyaffs2image/Makefile | 1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/mkyaffs2image/Makefile b/mkyaffs2image/Makefile
index 8c7d370..a71c20e 100644
--- a/mkyaffs2image/Makefile
+++ b/mkyaffs2image/Makefile
@@ -17,6 +17,7 @@
 CFLAGS =   -I/usr/include -O2 -Wall -DCONFIG_YAFFS_UTIL
 CFLAGS+=   -Wshadow -Wpointer-arith -Wwrite-strings -Wstrict-prototypes -Wmissing-declarations
 CFLAGS+=   -Wmissing-prototypes -Wredundant-decls -Wnested-externs -Winline
+CFLAGS+= -D_FILE_OFFSET_BITS=64
 
 CC=$(MAKETOOLS)gcc -m32
 
-- 
2.17.1

